app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,72 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    task = Task.new(task_params)
+
+    if task.save
+      redirect_to "/tasks"
+    end
+  end
+
+  def show
+    task_id = params[:id]
+    @task = Task.find(task_id)
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+
+  def update
+    task = Task.find(params[:id])
+    task.assign_attributes(task_params)
+
+    if task.save
+      redirect_to task
+    end
+  end
+
+  def complete
+    @task = Task.find(params[:id])
+    @task.completion_date = Date.today
+    @task.save
+
+    if @task.save
+      redirect_to tasks_path
+    end
+  end
+
+  def uncomplete
```

Comment:
You could consolidate these methods to be a single set of logic that can toggle the value: if the `completion_date` is `nil`, then set it. otherwise, set it to `nil`.